File: frontend/sistema-academico/src/app/shared/dto/disciplina-oferecida.dto.ts
import { SalaDto } from './sala.dto';
import { DisciplinaDto } from './disciplina.dto';
import { ProfessorDto } from './professor.dto';
import { DiaHoraDto } from './diaHora.dto';
import { CursaDisciplinaOferecidaDto } from './cursa-disciplina-oferecida.dto';

export interface DisciplinaOferecidaDto {
  disciplinaOferecidaId?: number;
  diasHoras?: DiaHoraDto[];
  professor?: ProfessorDto;
  disciplina?: DisciplinaDto;
  sala?: SalaDto;
  semestre?: number;
  turma?: number;
  ano?: number;
  disciplinasCursadas?: CursaDisciplinaOferecidaDto[];
}

File: backend/digitalhouse-pi/src/main/java/com/pi/dh/dto/DisciplinaOferecidaDTO.java
package com.pi.dh.dto;

import java.io.Serializable;
import java.util.List;

import lombok.Data;

@Data
public class DisciplinaOferecidaDTO implements Serializable {
	
	private static final long serialVersionUID = 1L;
	
	private Long disciplinaOferecidaId;
	private Integer semestre;
	private Short ano;
	private Integer turma;
	private List<DiaHoraDto> diasHoras;
	private ProfessorDTO professor;
	private DisciplinaDTO disciplina;
    private SalaDTO sala;
    //private List<CursaDisciplinaOferecidaDTO> disciplinasOferecidasCursadas;
    
}
